src/apprentice.c

```diff
@@ -338,7 +338,7 @@ static u16 GetRandomAlternateMove(u8 monId)
     const struct LevelUpMove *learnset;
     bool32 needTMs = FALSE;
     u16 moveId = MOVE_NONE;
-    bool32 shouldUseMove;
+    bool8 shouldUseMove;
```

Comment:
This can stay as `bool32`. Besides that looks good.

Reply:
Restored in https://github.com/rh-hideout/pokeemerald-expansion/pull/2233/commits/1c735e1abe0c872e661d68066c62cf2a0d4ad05f.